sql/core/src/test/scala/org/apache/spark/sql/execution/SubExprEliminationBenchmark.scala

```diff
@@ -45,7 +45,7 @@ object SubExprEliminationBenchmark extends SqlBasedBenchmark {
 
     withTempPath { path =>
       prepareDataInfo(benchmark)
-      val numCols = 1000
+      val numCols = 500
```

Comment:
@viirya, I had to reduce the number of predicates .. otherwise it fails (only) when you run it with `spark-submit` in JDK 11. It can be reproducible without using `Benchmarks` main.

Indeterministic exceptions. Either:

```
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$collectFirst$2(TreeNode.scala:231)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
	at scala.collection.immutable.List.foldLeft(List.scala:89)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$collectFirst$1(TreeNode.scala:231)
	at scala.Option.orElse(Option.scala:447)
	at org.apache.spark.sql.catalyst.trees.TreeNode.collectFirst(TreeNode.scala:231)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$collectFirst$3(TreeNode.scala:231)
	at scala.Option.orElse(Option.scala:447)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$collectFirst$2(TreeNode.scala:231)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
```

Or

```
        at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1679)
        at java.base/java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2464)
        at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2358)
        at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2196)
        at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1679)
        at java.base/java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2464)
        at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2358)
        at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2196)
        at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1679)
        at java.base/java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2464)
        at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2358)
```

Even with reducing, I think it shows the performance impact with the expression elimination.

I will update all the benchmark results according to the GA builds in a separate PR so I hope the current change is fine to you.